charts/devlake/templates/configmaps.yaml

```diff
@@ -58,20 +52,28 @@ data:
         secureJsonData:
           password: $MYSQL_PASSWORD
         editable: false
-#{{- else if eq .Values.option.database "pgsql" }}
-#  grafana-datasource.yml: |
-#    apiVersion: 1
-#    datasources:
-#      - name: postgres
-#        type: postgres
-#        url: $POSTGRES_URL
-#        database: $POSTGRES_DB
-#        user: $POSTGRES_USER
-#        secureJsonData:
-#          password: $POSTGRES_PASSWORD
-#        jsonData:
-#          sslmode: 'disable'
-#          postgresVersion: 1000 # support pgsql 10.0 and above
-#          timescaledb: false
 {{- end }}
 {{- end }}
+---
+{{- if .Values.grafana.oauthEnabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "devlake.fullname" . }}-grafana-oauth-config
+data:
+  {{- with .Values.grafana.oauthConfig }}
```

Comment:
Could this be optionally a secret? As far as I understand, it may contain sensitive data.

Reply:
At the moment, there is only a client_id and no client_secret (the value for the client secret is empty, see [here](https://github.com/apache/incubator-devlake-helm-chart/pull/125/files#diff-7fa69a54eb147fa46177bb22f6ea8f764514e74c78e9fe4847a8c1c9515482e7R169)); that's why all env vars are put here in a ConfigMap, rather than a Secret.

In the near future (May/June), there is a plan to support secrets store CSI driver / SecretProviderClass to sync secrets from secrets managers and create K8s Secrets automatically, but unfortunately, the priority isn't high enough to be implemented right now :)